packages/core/src/execution-engine/node-execution-context/utils/request-helper-functions.ts

```diff
@@ -309,6 +313,61 @@ export const createFormDataObject = (data: Record<string, unknown>) => {
 	return formData;
 };
 
+function copyRequestOptionsForLogs(
+	ctx: IAllExecuteFunctions,
+	requestOptions: IHttpRequestOptions,
+): IHttpRequestOptions | null;
+function copyRequestOptionsForLogs(
+	ctx: IAllExecuteFunctions,
+	requestOptions: IRequestOptions,
+): IRequestOptions | null;
+function copyRequestOptionsForLogs(
+	ctx: IAllExecuteFunctions,
+	requestOptions: IRequestOptions | IHttpRequestOptions,
+): IRequestOptions | IHttpRequestOptions | null {
+	const canBeLogged = !requestOptions.skipLogging && ctx.getNode().nodeDebugLogs;
+	if (!canBeLogged) {
+		return null;
+	}
+	// strip buffer bodies to prevent storing large data in memory
+	if (Buffer.isBuffer(requestOptions.body)) {
+		requestOptions = {
+			...requestOptions,
+			body: `Binary data got replaced with this text. Original was a Buffer with a size of ${
+				requestOptions.body.length
+			} bytes.`,
+		};
+	}
+	return deepCopy(requestOptions);
+}
+
+function createHttpRequestLogHandler(
+	ctx: IAllExecuteFunctions,
+	requestOptions: IRequestOptions | IHttpRequestOptions | null,
+	node: INode,
+	credentialsType: string,
+) {
+	return async () => {
+		// TODO: change to nodeDebugLogger and log request to server logs?
+		const ctxCompatible = 'sendMessageToUI' in ctx;
+		const canLog = node.nodeDebugLogs && ctx.getMode?.() === 'manual' && requestOptions !== null;
+		if (ctxCompatible && canLog) {
+			try {
+				let secrets: string[] = [];
+				const authKeys: IAuthDataSanitizeKeys = {};
+				if (credentialsType) {
+					const properties = ctx.getCredentialsProperties(credentialsType);
+					const credentials =
+						await ctx.getCredentials<ICredentialDataDecryptedObject>(credentialsType);
+					secrets = getSecrets(properties, credentials);
+				}
+				const sanitizedRequestOptions = sanitizeUiMessage(requestOptions, authKeys, secrets);
+				ctx.sendMessageToUI(sanitizedRequestOptions);
```

Comment:
you can use a new `nodeDebugLogger` as it would also use logger
e.g. `ctx.nodeDebugLogger(sanitizedRequestOptions, 'Request Options');`